### ISSUE TITLE
Complex number cast warnings from fft2 and fftn during backward pass

### ISSUE THREAD

**Issue by denjots** [pytorch/pytorch#59524]
## 🐛 Bug

The following warning is produced by torch.fft.fft2 and torch.fft.fftn (but not torch.fft.fft) when trying to take only the real
part of the complex output (for FFT token mixing) during backward pass but not forward:

[W Copy.cpp:240] Warning: Casting complex values to real discards the imaginary part (function operator())

## To Reproduce

Try to extract just the real portion of the return from fft2 and fftn during backward pass.

## Expected behavior

Behaviour should be consistent with torch.fft.fft and not produce a warning.

cc @ezyang @anjali411 @dylanbespalko @mruberry @Lezcano @peterbell10 @walterddr

**Comment by mruberry** [comment]
Hey @denjots, thanks for reporting this issue. Can you provide a short script that shows the issue you're having? Like a few lines of Python?

**Comment by denjots** [comment]
Sure, here you go... same issue when you replace fftn with fft2 - but not fft.

```
import torch

x = torch.rand(10,10,10,requires_grad=True)
x = torch.fft.fftn(x)
x.mean().backward()
```

**Comment by mruberry** [comment]
Thank you, @denjots.

@peterbell10, would you mind taking a look at this?

**Comment by peterbell10** [comment]
This will be fixed by #60021.
